python/ray/worker.py

```diff
@@ -1276,16 +1278,19 @@ def connect(node,
             worker.logger_thread.start()
 
     if mode == SCRIPT_MODE:
-        # Add the directory containing the script that is running to the Python
-        # paths of the workers. Also add the current directory. Note that this
-        # assumes that the directory structures on the machines in the clusters
-        # are the same.
-        script_directory = os.path.abspath(os.path.dirname(sys.argv[0]))
-        current_directory = os.path.abspath(os.path.curdir)
-        worker.run_function_on_all_workers(
-            lambda worker_info: sys.path.insert(1, script_directory))
-        worker.run_function_on_all_workers(
-            lambda worker_info: sys.path.insert(1, current_directory))
+        # In client mode, if we use runtime env, then it'll be taken care of
+        # automatically.
+        if not job_config.client_job and job_config.get_runtime_env_uris():
+            # Add the directory containing the script that is running to the Python
+            # paths of the workers. Also add the current directory. Note that this
+            # assumes that the directory structures on the machines in the clusters
+            # are the same.
+            script_directory = os.path.abspath(os.path.dirname(sys.argv[0]))
+            current_directory = os.path.abspath(os.path.curdir)
+            worker.run_function_on_all_workers(
+                lambda worker_info: sys.path.insert(1, script_directory))
+            worker.run_function_on_all_workers(
+                lambda worker_info: sys.path.insert(1, current_directory))
         # TODO(rkn): Here we first export functions to run, then remote
```

Comment:
@edoakes  could you take another look at this part? client-server will fail since the curdir is invalid (it got GCed once there is no instance using this). And for client-server, it's set up in another two places (please check my comment).